### PR TITLE
Make Beam version validation for DataflowTemplates easier

### PR DESCRIPTION
**Pull request by Polber**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,9 +46,9 @@
     <jacoco.version>0.8.8</jacoco.version>
 
     <!-- Beam and linked versions -->
-    <!-- Also update beam version in requirements.txt throughout the repo -->
-    <beam.version>2.56.0</beam.version>
-    <jackson.version>2.14.1</jackson.version>
+    <beam-java.version>2.56.0</beam-java.version>
+    <beam-python.version>${beam-java.version}</beam-python.version>
```
Comment (Abacn) [inline pom.xml:50]: imo we can keep `${beam.version}` and create `<beam-python.version>`. This minimizes the footprint of change and avoid any unexpected / missing places need to make change.

**Review by Abacn — COMMENTED** [review]
Thanks! Had a few comments to help reduce the scope of change needed

File: python/src/test/java/com/google/cloud/teleport/templates/python/YAMLTemplateIT.java [hunk 2]
```diff
@@ -105,6 +106,9 @@ private void runYamlTemplateTest(
       Function<PipelineLauncher.LaunchConfig.Builder, PipelineLauncher.LaunchConfig.Builder>
           paramsAdder)
       throws IOException {
+    boolean useStagedJars =
```
[the remark this answers is not in the document]
Comment (Polber) [reply python/src/test/java/com/google/cloud/teleport/templates/python/YAMLTemplateIT.java:109]: Good point, I'll refactor that

**Comment by Polber** [comment]
@Abacn addressed your comments

**Review by Abacn — APPROVED** [review]
thank you!

**Pull request merged**